packages/discovery-react-components/src/components/SearchFacets/components/FacetsGroups/MultiSelectFacetsGroup.tsx

```diff
@@ -56,6 +56,12 @@ export const MultiSelectFacetsGroup: FC<MultiSelectFacetsGroupProps> = ({
     <>
       {facets.map(facet => {
         const text = get(facet, facetsTextField, '');
+        const matchingResults = get(facet, 'matching_results');
+        let fullText = text;
+        // To prevent showing the number of Dynamic Facets until the API supports this
+        if (matchingResults > 0) {
```

Comment:
One reason we didn't implement this initially is due to how we end up going through and selecting facets. I would try using it a bit with multiple facets to see what ends up showing for the matching results after selecting across facet groups and make sure that doesn't look strange (specifically, whether or not the numbers change or go to 0 when we would/wouldn't expect them to)

Reply:
In my experience playing with the facets, the matching results value never changes. Each selected facet adds its matching results to the list of documents shown--it does not filter down.

Reply:
@H-Fin that's something i would bring up with design -> do we want those numbers to change?

the implementation we went with was one that allowed you to expand selections after 2 or more facet groups are selected.

the alternative is to filter down the list of available facets to the ones returning on the request to indicate how your filter affected the result set. the downside of this approach is that exploratory faceting is mostly eliminated without undoing selections (no way to say "i want "x" OR "y" instead of just "x" AND "y")

Reply:
also - i think we order by these `matching_results` but i faintly remember also bubbling the selected ones to the top, does that happen? does it look weird if the numbers are out of order?

Reply:
We don't bubble selected ones to the top. As of right now, after each search query, the non-dynamic facets do not change order or appear/disappear based off facet filtering.

Reply:
got it - must be the dynamic facets that do that

Reply:
I am investigating what the expected behavior is of the other facets though and will update accordingly.

Reply:
A separate issue will address facet groups (#1352). I added a gif to this PR so people can see the current behavior more easily.